### ISSUE TITLE
Is it possible to verify if all gems in a Gemfile.lock exist at https://rubygems.org? - i.e. enhance `bundle check`

### ISSUE THREAD

**Issue by stwr667** [rubygems/rubygems#4487]
### Background
I recently updated my elastic beanstalk platform version, and it failed due to the following error:

> Your bundle is locked to mimemagic (0.3.3), but that version could not be found in any of the sources listed in your Gemfile. If you haven't changed sources, that means the author of mimemagic (0.3.3) has removed it. You'll need to update your bundle to a version other than mimemagic (0.3.3) that hasn't been removed in order to install.

It was difficult to debug due to all the complexities of a managed deploy system.

The reason it failed is that the author of mimemagic has removed version 0.3.3, and so I needed to update the gem to 0.3.6. (see gem versions history here: https://rubygems.org/gems/mimemagic/versions)

### Feature Suggestion
Given the above, it would be incredibly useful to be able to run a command that scans the gems and versions in your `Gemfile.lock` and hits https://rubygems.org to verify that they all exist and are still available?

It would be similar to what `bundle check` does, but instead of looking for the gems on your local machine, it would look for them at the remote source. Something like `bundle check --remote` or `bundle check --source`?

**Comment by deivid-rodriguez** [comment]
Not opposed to adding something like what you proposed, but isn't the current error message quite accurate?

**Comment by stwr667** [comment]
@deivid-rodriguez absolutely, the current error message is accurate. But if I have 100 gems in my gemfile, currently, the only way to know that e.g. mimemagic is going to fail during a platform upgrade is to remove all the installed gems locally and then try and re-install them (and get the above error), which is slow and painful.
It would be great to be able to simply run `bundle check --remote` which doesn't install/uninstall anything, but can verify for you that all gems exist at your specified remote source, with their respective versions listed in `Gemfile.lock`.

**Comment by deivid-rodriguez** [comment]
Sue, that makes sense, yeah. Note though that the only way to make sure that all your locked gems can actually be installed is... installing them, since gems could be yanked at any time.

**Comment by deivid-rodriguez** [comment]
Did you try `bundle install --redownload`, by the way?

**Comment by stwr667** [comment]
I didn't know about `bundle install --redownload`. When I ran that, my bundle installed successfully, because it automatically picked mimemagic version `0.3.6` instead of the `0.3.3` in the `Gemfile.lock`. I don't specifically have mimemagic in my Gemfile - it's a nested dependency of another gem I have (rails -> activerecord -> activestorage -> marcel -> mimemagic (~> 0.3.2)). So to reproduce the above error I needed to run: `bundle install --deployment --redownload` (because the `--deployment` switch forces the ignoring of the `~>` which would normally happily upgrade patch versions).

That's definitely the best alternative to this feature request, since it's a 1 line command to verify it all. Thanks for that suggestion.

So I guess the benefits of this feature suggestion are now limited to:
- Faster performance (just checking availability, rather than doing any installing)
- Leaving your local gemsets untouched

Thanks @deivid-rodriguez!

**Comment by rubyFeedback** [comment]
> The reason it failed is that the author of mimemagic has removed version 0.3.3

That example about mimemagic hit the news in the last days. ;)

Guess this may encourage thoughts what to do when gems are yanked.
Authors should have the right to yank at any moment in time (just
as a github repository can be removed too), but downstream users
may be affected too so perhaps a "transition cache" could be used
for examples like this. Like "ghost gems", but with a clear providsion
that they may not be offered forever. Like some automatic time
passed before they become unavailable, just as mitigation help
against yanked gems.
